app/scripts/controllers/permissions/permissions.js

```diff
@@ -94,6 +95,35 @@ class PermissionsController {
     })
   }
 
+  /**
+   * @param {string} origin - The external domain id.
+   * @param {Array<string>} requestedPlugins - The names of the requested plugin permissions.
+   */
+  async handleInstallPlugins (origin, requestedPlugins) {
+
+    const existingPerms = this.permissions.getPermissionsForDomain(origin).reduce(
+      (acc, p) => {
+        acc[p.parentCapability] = true
+        return acc
+      }, {}
+    )
+
+    requestedPlugins.forEach(p => {
+      if (!existingPerms[p]) {
```

Comment:
This seems to imply that the permission to install a plugin is the same as its `parentCapability`? Does this mean the permission request would look like this?:
```javascript
await ethereum.send('wallet_requestPermissions', [{
  'ens://ipfs.snap.eth': {},
}])
await ethereum.send('wallet_plugins_installPlugins', ['ens://ipfs.snap.eth'])
```

I would feel a little weird allowing plugin domains to be reflected as top-level permissions, since it's not clear that permission is a plugin.

Maybe instead we should just have a nested capability request, something more like:
```javascript
await ethereum.send('wallet_requestPermissions', [{
  'wallet_plugins_installPlugins': {
    'plugins: {
      'ens://ipfs.snap.eth': {},
    }
  },
}])
await ethereum.send('wallet_plugins_installPlugins', ['ens://ipfs.snap.eth'])
```

Or we could build it as caveats:
```javascript
await ethereum.send('wallet_requestPermissions', [{
  'wallet_plugins_installPlugins': {
    caveats: [
      { type: 'filterParams', value: [ { type: 'or', value: ['ens://ipfs.snap.eth', 'ens://plasma.eth'] }],
    ]
  },
}])
await ethereum.send('wallet_plugins_installPlugins', ['ens://ipfs.snap.eth'])
```

Anyways, I guess I just want some clarity on exactly what the interaction looks like.

Reply:
I'm drafting a longer response to this, but as far as the plugin permission names go, they are still `wallet_plugin_`-prefixed. That prefix is removed here, in `PluginsController.processRequestedPlugins`:
https://github.com/MetaMask/metamask-snaps-beta/blob/b65e2f37a462d01fe4ae2913a18d5f64a162e11c/app/scripts/controllers/plugins.js#L126-L128

Reply:
Resolved in #94